p2p/net/swarm/swarm.go

```diff
@@ -82,6 +90,8 @@ func NewSwarm(ctx context.Context, listenAddrs []ma.Multiaddr,
 	s.cg.SetTeardown(s.teardown)
 	s.SetConnHandler(nil) // make sure to setup our own conn handler.
 
+	prom.MustRegisterOrGet(peersTotal)
```

Comment:
That's for another PR, imo. Right now this basically counts "total number of peers over all swarms", which is a good simple metric to get started with.

If the swarm itself is an interesting dimension of the peers count, we can replace the Counter with a CounterVec, which allows supplying labels (i.e. dimensions) with each Inc()/Dec() call. [More info on labels](http://prometheus.io/docs/concepts/data_model/).

> CounterVec is a Collector that bundles a set of Counters that all share the same Desc, but have different values for their variable labels. This is used if you want to count the same thing partitioned by various dimensions (e.g. number of HTTP requests, partitioned by response code and method). Create instances with NewCounterVec. 

```
peersTotal.GetMetricWithLabelValues(Labels{"swarm": "foo"}).Inc(1)
```

Labels should only be used with dimensions that are more or less bounded. For example, a unix process ID is a bad label.